src/checker.py

```diff
@@ -249,15 +253,24 @@ def _update_manifest(self, path, datas, changes):
                 message = "Update {} to {}".format(
                     data.filename, data.new_version.version
                 )
+                last_version = data.new_version.version
             else:
                 message = "Update {}".format(data.filename)
 
             changes[message] = None
 
         if changes:
-            print("Updating {}".format(path))
+            log.info("Updating %s", path)
             self._dump_manifest(path)
 
+            appdata = os.path.splitext(path)[0] + ".appdata.xml"
+            if last_version is not None and os.path.exists(appdata):
+                # This assumes that the last changed source for which we can detect a
+                # new version is the interesting one – this may or may not be true!
+                # TODO: extract date in checkers
+                release_date = dt.date.today()
```

Comment:
hmm, ideally we should have a way to get the latest release info (version+date) for a "top" manifest (from path) instead, but while we don't support that I think using the last changed source is Ok, although I can imagine a situation where we have multiple external data/sources using different versions.

On possible option would be to grow support for specialized manifest checkers (alongside extra-data checkers), with each manifest checker running the external data checkers for that component + providing release info.

Reply:
Agreed. I realised that `date.today()` is actually not good enough: the branch name used depends on the contents of the tree, so it would change every day and file a duplicate PR until one is merged. Unfortunately the Debian repo format doesn't include timestamps, so I make a `HEAD` request against the package URL and use its timestamp.

Reply:
Another option would be to extract the date from the package changelog itself (which may require fetching/extracting the package if python apt doesn't provide that) and for firefox we could get the date from `LAST_RELEASE_DATE` returned on https://product-details.mozilla.org/1.0/firefox_versions.json.

Reply:
`python-apt` has some support for fetching changelogs for Debian and Ubuntu, but we don't have a changelog server. https://salsa.debian.org/apt-team/python-apt/blob/master/apt/package.py#L1245-1417

The other option would be to fetch the package (or just the relevant half of the source package – if we have a source package, that is!) and parse the changelog (if present). Personally, I think we can make do with the `HEAD` request for now. I'll add a TODO about this.

---

https://product-details.mozilla.org/1.0/firefox_versions.json has `"LAST_RELEASE_DATE": "2019-07-09"` but https://www.mozilla.org/en-US/firefox/68.0.2/releasenotes/ says:

> Version 68.0.2, first offered to Release channel users on August 14, 2019

So I don't think that's going to fly. (I'll add a note about this.) However, we can and should propagate the version number from that file into the `ExternalFile` object. I'll make that change.

While looking at this, I realised that we currently fetch all the files for Firefox, but we don't need to – we get their paths from `SHA256SUMS` so we have those, and we can just make a `HEAD` request to get the size. That's another task though.